Lib/configparser.py

```diff
@@ -1027,36 +1056,61 @@ def _read(self, fp, fpname):
                         # So sections can't start with a continuation line
                         optname = None
-                    # no section header in the file?
+                    # no section header?
                     elif cursect is None:
                         raise MissingSectionHeaderError(fpname, lineno, line)
                     # an option line?
                     else:
-                        mo = self._optcre.match(value)
+                        indent_level = cur_indent_level
+                        # is it a section header?
+                        mo = self.SECTCRE.match(value)
                         if mo:
-                            optname, vi, optval = mo.group('option', 'vi', 'value')
-                            if not optname:
-                                e = self._handle_error(e, fpname, lineno, line)
-                            optname = self.optionxform(optname.rstrip())
-                            if (self._strict and
-                                (sectname, optname) in elements_added):
-                                raise DuplicateOptionError(sectname, optname,
-                                                        fpname, lineno)
-                            elements_added.add((sectname, optname))
-                            # This check is fine because the OPTCRE cannot
-                            # match if it would set optval to None
-                            if optval is not None:
-                                optval = optval.strip()
-                                cursect[optname] = [optval]
+                            sectname = mo.group('header')
+                            if sectname in self._sections:
+                                if self._strict and sectname in elements_added:
+                                    raise DuplicateSectionError(sectname, fpname,
+                                                                lineno)
+                                cursect = self._sections[sectname]
+                                elements_added.add(sectname)
+                            elif sectname == self.default_section:
+                                cursect = self._defaults
                             else:
-                                # valueless option handling
-                                cursect[optname] = None
+                                cursect = self._dict()
+                                self._sections[sectname] = cursect
+                                self._proxies[sectname] = SectionProxy(self, sectname)
+                                elements_added.add(sectname)
+                            # So sections can't start with a continuation line
+                            optname = None
+                        # no section header in the file?
+                        elif cursect is None:
+                            raise MissingSectionHeaderError(fpname, lineno, line)
+                        # an option line?
```

Comment:
This code is unreachable and duplicates code above. See https://github.com/python/cpython/commit/63ddbfee3afe8e1b2e6be3086c87cfdd0d73c2b9. This finding reinforces my understanding that the code is too complex to reason about to make changes safely. If we remove this change now, it'll break all of the refactoring work I've done in that branch, so let's leave it here.

Reply:
Duplication will be addressed in #117348.